buildSrc/src/main/kotlin/com/akuleshov7/Versions.kt

```diff
@@ -8,4 +8,5 @@ object Versions {
     const val JUNIT = "5.7.1"
```

Comment:
Why you did not remove Junit then?

Reply:
Looks like Junit still remains @ILikeYourHat what was the idea?

Reply:
Migrating to Kotest is a big change, and my idea was to migrate step by step, PR by PR, to make CR easier and avoid possible conflicts. However I can migrate all tests at once in this PR, if you prefer that way. It is up to you :)

Reply:
If we do a migration - it is difficult to keep both Unit and Kotest :( 
Better to migrate everything, sorry 🙏

Reply:
@akuleshov7 it took a while, but here you have it: everything migrated to Kotest :)

Please note that the actual runner for jvm is still JUnit5. Kotest also have an jvm test runner, but the test format is very different there: https://kotest.io/docs/framework/writing-tests.html. I know, the issue mentions also the change of the runner, but in my opinion:
- the main benefit are assertions, and those are migrated
- things like parametrized test, user friendly names of tests etc. can be achieved also in JUnit5 runner
- this PR is big enough, and I don't have enough energy for increasing the scope